pkg/project/admission/requestlimit/admission.go

```diff
@@ -0,0 +1,156 @@
+	projectapi "github.com/openshift/origin/pkg/project/api"
+	projectcache "github.com/openshift/origin/pkg/project/cache"
+)
+
+func init() {
+	admission.RegisterPlugin("ProjectRequestLimit", func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig, err := readConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		return NewProjectRequestLimit(pluginConfig)
+	})
+}
+
+func readConfig(reader io.Reader) (*ProjectRequestLimitConfig, error) {
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
+		return &ProjectRequestLimitConfig{}, nil
+	}
+
+	configBytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	config := &ProjectRequestLimitConfig{}
+	err = configlatest.ReadYAML(configBytes, config)
+	if err != nil {
+		return nil, err
+	}
+	errs := ValidateProjectRequestLimitConfig(config)
+	if len(errs) > 0 {
+		return nil, errutil.NewAggregate(errs)
+	}
+	return config, nil
+}
+
+type projectRequestLimit struct {
+	*admission.Handler
+	client client.Interface
+	config *ProjectRequestLimitConfig
+	cache  *projectcache.ProjectCache
+}
+
+// ensure that the required Openshift admission interfaces are implemented
+var _ = oadmission.WantsProjectCache(&projectRequestLimit{})
+var _ = oadmission.WantsOpenshiftClient(&projectRequestLimit{})
+var _ = oadmission.Validator(&projectRequestLimit{})
+
+// Admit ensures that only a configured number of projects can be requested by a particular user.
+func (o *projectRequestLimit) Admit(a admission.Attributes) (err error) {
+	if a.GetResource() != "projectrequests" {
+		return nil
+	}
+	if _, isProjectRequest := a.GetObject().(*projectapi.ProjectRequest); !isProjectRequest {
+		return nil
+	}
+	userName := a.GetUserInfo().GetName()
+	projectCount, err := o.projectCountByRequester(userName)
+	if err != nil {
+		return err
+	}
+	maxProjects, hasLimit, err := o.maxProjectsByRequester(userName)
+	if err != nil {
+		return err
+	}
+	if hasLimit && projectCount >= maxProjects {
+		return admission.NewForbidden(a, fmt.Errorf("user %s cannot create more than %d project(s).", userName, maxProjects))
+	}
+	return nil
+}
+
+// maxProjectsByRequester returns the maximum number of projects allowed for a given user, whether a limit exists, and an error
+// if an error occurred. If a limit doesn't exist, the maximum number should be ignored.
+func (o *projectRequestLimit) maxProjectsByRequester(userName string) (int, bool, error) {
+	// prevent a user lookup if no limits are configured
+	if len(o.config.Limits) == 0 {
+		return 0, false, nil
+	}
+
+	user, err := o.client.Users().Get(userName)
+	if err != nil {
+		return 0, false, err
+	}
+	userLabels := labels.Set(user.Labels)
+
+	for _, limit := range o.config.Limits {
+		selector := labels.Set(limit.Selector).AsSelector()
+		if selector.Matches(userLabels) {
+			if limit.MaxProjects == nil {
+				return 0, false, nil
+			}
+			return *limit.MaxProjects, true, nil
```

Comment:
Not sure that we have a use case for that or that selectors are going to be that complex. Right now it's relying on the order of selectors in the config and, for what we want this for, imho it's sufficient.